lib/use-realtime-price.tsx

```diff
@@ -0,0 +1,121 @@
+import { gql, SubscriptionResult } from "@apollo/client"
+import * as React from "react"
+import {
+  RealtimePriceWsSubscription,
+  useRealtimePriceWsSubscription,
+} from "../lib/graphql/generated"
+import { useDisplayCurrency } from "../lib/use-display-currency"
+
+gql`
+  subscription realtimePriceWs($currency: DisplayCurrency!) {
+    realtimePrice(input: { currency: $currency }) {
+      errors {
+        message
+      }
+      realtimePrice {
+        timestamp
+        btcSatPrice {
+          base
+          offset
+        }
+        usdCentPrice {
+          base
+          offset
+        }
+        denominatorCurrency
+      }
+    }
+  }
+`
+
+const useRealtimePrice = (
+  currency: string,
+  onSubscriptionDataCallback?: (
+    subscriptionData: SubscriptionResult<RealtimePriceWsSubscription, any>,
+  ) => void,
+) => {
+  const priceRef = React.useRef<number>(0)
+  const { formatCurrency } = useDisplayCurrency()
+  const hasLoaded = React.useRef<boolean>(false)
+
+  const { data } = useRealtimePriceWsSubscription({
+    variables: { currency },
+    onSubscriptionData({ subscriptionData }) {
+      if (onSubscriptionDataCallback) onSubscriptionDataCallback(subscriptionData)
+    },
+  })
+
+  React.useEffect(() => {
+    if (data && !hasLoaded.current) {
+      // Subscription data has loaded for the first time
+      hasLoaded.current = true
+    }
+  }, [data])
+
+  const conversions = React.useMemo(
+    () => ({
+      satsToCurrency: (sats: number, display: string, fractionDigits: number) => {
+        const convertedCurrencyAmount =
+          fractionDigits === 2 ? (sats * priceRef.current) / 100 : sats * priceRef.current
```

Comment:
Im not following this logic. What if fractionDigits is 3, do you not adjust then?

Reply:
Thats computed in the component so i didn't have to depend on downloading the entire currencyList to get the currency metadata. so if the fractionDigits is 3 it computes it here: https://github.com/GaloyMoney/galoy-pay/pull/434/files#diff-b64a902b7dc296d175efc931c6439ed001de3251ff67466bf9d3e3006f5faefcR160 and here https://github.com/GaloyMoney/galoy-pay/pull/434/files#diff-b64a902b7dc296d175efc931c6439ed001de3251ff67466bf9d3e3006f5faefcR168